### PR TITLE
Add support to ingest inline attachments

### PR DESCRIPTION
**Pull request by Swatinem**
For individual attachments, it is now possible to pass the contents within the `data` property directly instead of needing to do a processing attachments store roundtrip via `attachment_chunk` messages.

### CODE REVIEW

File: src/sentry/ingest/consumer/processors.py [hunk 1]
```diff
@@ -280,13 +280,17 @@ def process_individual_attachment(message: IngestMessage, project: Project) -> N
...
-        key=cache_key, type=attachment.pop("attachment_type"), **attachment
-    )
-    if attachment.type not in ("event.attachment", "event.view_hierarchy"):
-        logger.error("invalid individual attachment type: %s", attachment.type)
+    attachment_type = attachment.pop("attachment_type")
+    if attachment_type not in ("event.attachment", "event.view_hierarchy"):
+        logger.error("invalid individual attachment type: %s", attachment_type)
+
+        # clear invalid attachments from the cache directly
+        attachment.delete()
```
Comment (loewenheim) [inline src/sentry/ingest/consumer/processors.py:288]: The comment says `clear invalid attachments from the cache directly`, but `attachment` is an entry in a mapping (`message`) here. Can you clarify what cache is being referenced here?

File: src/sentry/ingest/consumer/processors.py [hunk 2]
```diff
@@ -280,13 +280,17 @@ def process_individual_attachment(message: IngestMessage, project: Project) -> N
...
         return
 
+    # NOTE: `get_from_chunks` will avoid the cache if `attachment` contains `data` inline
```
Comment (loewenheim) [inline src/sentry/ingest/consumer/processors.py:291]: Is this already implemented?

Comment (Swatinem) [reply src/sentry/ingest/consumer/processors.py:291]: it is, yes :-)

File: src/sentry/ingest/consumer/processors.py (hunk 1 above)
Comment (Swatinem) [reply src/sentry/ingest/consumer/processors.py:288]: indeed, the variable was wrongly shadowed. I moved the code around and gave each its own unique name.

**Review by loewenheim — APPROVED** [review]

**Pull request merged**